tests/lib/SimpleSAML/Utils/Config/MetadataTest.php

```diff
@@ -259,4 +259,44 @@ public function testIsHiddenFromDiscovery()
             ],
         ]));
     }
+
+    
+    /**
+     * Test \SimpleSAML\Utils\Config\Metadata::parseNameIdPolicy().
+     */
+    public function testParseNameIdPolicy()
+    {
+        // Test null or unset
+        $nameIdPolicy = null;
+        $this->assertEquals(Metadata::parseNameIdPolicy($nameIdPolicy), ['Format' => \SAML2\Constants::NAMEID_TRANSIENT]);
+
+        // Test false
+        $nameIdPolicy = false;
+        $this->assertEquals(Metadata::parseNameIdPolicy($nameIdPolicy), null);
+
+        // Test string
+        $nameIdPolicy = 'urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress';
+        $this->assertEquals(Metadata::parseNameIdPolicy($nameIdPolicy), ['Format' => 'urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress']);
+
+        // Test array
+        $nameIdPolicy = [
+            'Format' => 'urn:oasis:names:tc:SAML:1.1:nameid-format:persistent',
+            'AllowCreate' => false
+        ];
+        $this->assertEquals(Metadata::parseNameIdPolicy($nameIdPolicy), [
+            'Format' => 'urn:oasis:names:tc:SAML:1.1:nameid-format:persistent',
+            'AllowCreate' => false
+        ]);
```

Comment:
This could be simplified:

```php
        $nameIdPolicy = [
            'Format' => 'urn:oasis:names:tc:SAML:1.1:nameid-format:persistent',
            'AllowCreate' => false
        ];
        $this->assertEquals($nameIdPolicy, Metadata::parseNameIdPolicy($nameIdPolicy));
```